docs/python/formatting.md

```diff
@@ -14,7 +14,32 @@ Formatting makes source code easier to read by human beings. By enforcing partic
 
 [Linting](/docs/python/linting.md) helps to prevent errors by analyzing code for common syntactical, stylistic, and functional errors and unconventional programming practices. Although there is a little overlap between formatting and linting, the two capabilities are complementary.
 
-The Python extension supports source code formatting through formatter extensions, such as [autopep8](https://marketplace.visualstudio.com/items?itemName=ms-python.autopep8) and [Black Formatter](https://marketplace.visualstudio.com/items?itemName=ms-python.black-formatter).
+## Choose a formatter
+
+Install the formatting tool of your choice from the VS Code [Marketplace](https://marketplace.visualstudio.com/vscode).
+
+Microsoft publishes the following formatting extensions:
+
+| Formatter | Extension                                                                       |
+| ------ | ------------------------------------------------------------------------------- |
+| autopep8 | [https://marketplace.visualstudio.com/items?itemName=ms-python.autopep8](https://marketplace.visualstudio.com/items?itemName=ms-python.autopep8)            |
+| Black formatter | [https://marketplace.visualstudio.com/items?itemName=ms-python.black-formatter](https://marketplace.visualstudio.com/items?itemName=ms-python.black-formatter)            |
+
+Formatter extensions offered by the community:
+
+| Formatter | Extension                                                              |
+| ------ | ---------------------------------------------------------------------- |
+| Ruff   | [https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff](https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff)   |
+| yapf   | [https://marketplace.visualstudio.com/items?itemName=eeyore.yapf](https://marketplace.visualstudio.com/items?itemName=eeyore.yapf) |
+
+Furthermore, below are formatter extensions that support import sorting:
+
+| Formatter | Extension                                                              |
+| ------ | ---------------------------------------------------------------------- |
+| Ruff   | [https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff](https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff)   |
+| isort   | [https://marketplace.visualstudio.com/items?itemName=ms-python.isort](https://marketplace.visualstudio.com/items?itemName=ms-python.isort) |
```

Comment:
These also need instructions on how to set them as the import organizer.

```
{
  "[python]": {
    "editor.codeActionsOnSave": {
      "source.organizeImports.ruff": true
    }
  }
}
```

Reply:
Pushed changes for this under the **Set a default formatter** section with an example for Ruff configuration. Let me know if the wording/references for the scope/settings hierarchy is correct, I'm not very familiar with the naming of these components.